FormSubmissionAndValidation2/src/components/CheckBox.jsx

```diff
@@ -0,0 +1,33 @@
+import PropTypes from "prop-types";
+import { ErrorMessage } from "./ErrorMessage";
+import { userFormConfig } from "../helper/userHelper";
+import React from "react";
+
+export const CheckBox = React.memo(function CheckBox({fieldId, value, errorMessage, updateValue}) {
+  const {label, message} = userFormConfig[fieldId];
+  
+  const checked = value ? "checked" : ""
+  return <>
+    <div className="checkbox">
+      <input 
+        id={label}
+        type="checkbox" 
+        data-inputfield="receiveNotification" 
+        value={checked}
+        onChange={event => updateValue(fieldId, event.target.checked)}
+        checked={value || false}
+        />
+      <label className="checkboxLabel" htmlFor={label}>{label}</label>
+      <br />
+    </div>
+    <ErrorMessage message={errorMessage} />
+    <div className="note">{message}</div>
+  </>
+})
+
+CheckBox.propTypes = {
```

Comment:
Any of these proptypes are required?

Also, to clarify, you have read up on typescript interfaces and definitions, right?
I assume this exercise was done before that?

Reply:
yes I think I haven't added isRequired at any place. Will check and add at all the places

Reply:
yes, these exercises were submitted before I read TypeScript